server/channel_player.go

```diff
@@ -661,6 +661,19 @@ func (server ChannelServer) playerMoveInventoryItem(conn mnet.Client, reader mpa
 	}
 }
 
+func (server ChannelServer) playerUseInventoryItem(conn mnet.Client, reader mpacket.Reader) {
+	plr, err := server.players.getFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	slot := reader.ReadInt16()
+	itemid := reader.ReadInt32()
+
+	plr.UseItem(slot, itemid)
```

Comment:
I think the item usage should be something like:
```
item, err := plr.TakeItem()

if err != nil {
}

item.Apply(plr)
```

This way item logic would be within the item package instead of bloating the player package.

Reply:
I like the idea of making more reusable code; however, wouldn't that create an import cycle? Specifically creating a function in the item package that accepts type *player.Data . I believe that would be needed to alter the players stats when an item is applied.

Reply:
The `Apply` function should take an interface instead of the `*player.Data` type. This has the added benefit of making it easier to unit test items if that ever gets done, as the unit test would create a dummy type that matches the interface.

Reply:
I'm not sure I understand what you are wanting out of the item.Apply function. If an interface is passed rather than a struct we will not be able to interact with the player's data. 

As an example for the following function
`func (v *Data) ApplyUseItem(plr interface{}) {
}`

We would not be able to do something such as this to apply an HP potion because an interface is ambiguous therefore we cannot make the call to plr.hp . 
`	if v.hp > 0 {
		if (plr.hp - v.hp) < 0 {
			plr.hp = 0
		} else {
			plr.hp = plr.hp - v.hp
		}
	}`

Can you explain a little more what you are envisioning specifically for the item.Apply() function?

Reply:
What I mean by interface is something like:
```
type player interface {
  NoChange()
  GiveHP(int16)
  GiveMP(int16)
}

func (v *Data) ApplyUseItem(plr player) { }
```

instance.go in the field package does a similar thing - at the top of the file you will see a player interface defined

Reply:
Ah okay perfect, that will work. I will refactor it and update PR.